File: src/main/resources/static/legalEntity/legalEntityService.js
var services = angular.module('legalEntity.services',['ngResource']);

services.service('legalEntityService',['$http',function($http){

	this.findAllBusinessActivities = function(){
		return $http.get('/businessActivityCode')
	}
	
	this.findAllPlaces = function(){
		return $http.get("/place");
	}
	
	this.findAll = function(){
		return $http.get("/legalEntity");
	}

	this.search = function(legalEntity){
		return $http.post("/legalEntity/search",legalEntity);
	}
	this.delete = function(id){
		return $http.delete("/legalEntity/"+id+"/")
	}
	
	this.nextFilterBussinesActivity = function(id){
		return $http.get("/legalEntity/nextFilterBussinesActivity/" + id);
	}
	
	this.save = function(legalEntity) {
		return $http.post("/legalEntity",legalEntity);
	}
	this.update = function(legalEntity){
		return $http.put("/legalEntity/update", legalEntity);
	}
	
}])

File: src/main/java/bank/legalEntity/LegalEntityService.java
package bank.legalEntity;

import java.util.List;

import bank.bank.Bank;

public interface LegalEntityService {

	public List<LegalEntity> findAll();
	public List<LegalEntity> search(LegalEntity legalEntity);
	public LegalEntity findOne(Long id);
	public void delete(Long id);
	public LegalEntity save(LegalEntity legalEntity);
}
